includes/class-wc-stripe-customer.php

```diff
@@ -410,12 +412,29 @@ public function get_sources() {
 				'GET'
 			);
 
+			if ( empty( $response->error ) && is_array( $response->data ) ) {
+				$sources = $response->data;
+			} else {
+				$sources = [];
+			}
+
+			// Add any payment methods that might have been saved to the customer.
+			$response = WC_Stripe_API::request(
+				[
+					'limit'    => 100,
+					'customer' => $this->get_id(),
+					'type'     => 'card',
+				],
+				'payment_methods',
+				'GET'
+			);
+
 			if ( ! empty( $response->error ) ) {
 				return [];
 			}
 
-			if ( is_array( $response->data ) ) {
-				$sources = $response->data;
+			if ( empty( $response->error ) && is_array( $response->data ) ) {
+				$sources = array_merge( $sources, $response->data );
```

Comment:
> Is the idea to merge both Sources and PaymentMethods into the user's payment tokens?

Yes.

> The /payment_methods API will return both pm_'s and src_'s, so this will result in a list of repeated payment methods. We should request only /payment_methods I guess.

Oh, I didn't know that! I thought they'd be separate since that's how most of the API operates 🤔 

Are you sure about this? I see that the documentation always refers to "PaymentMethods" 🤔 

> I was checking Stripe's API on the above and apparently they recommend listing the PaymentMethods from the Customer instead

Switched to that in 67a4f8e778f6fc1d844a4fd9982d5ee86c738de1, assuming you're correct about the API returning both Payment Methods and Sources. If that's not the case we can revert that commit 👍 

> Also, filtering it for type: card won't return the sepa_debit sources from the non-UPE checkout. As far as I've tested it, it works fine without the type: card.

Copied from my previous PR without thinking about it. I've removed it now 🙂

Reply:
Or maybe we should ask Stripe to confirm, just in case we might be missing some edge case where the API doesn't always return a Source.

Reply:
Awesome, thank you! I'll keep it as is then 🙂 

> Or maybe we should ask Stripe to confirm, just in case we might be missing some edge case where the API doesn't always return a Source.

Hmm, probably not a bad idea 🤔 
I'll send them a question in Slack tomorrow 👍